core/src/main/kotlin/ch/ergon/dope/resolvable/expression/CaseExpression.kt

```diff
@@ -0,0 +1,132 @@
+package ch.ergon.dope.resolvable.expression
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.ValidType
+
+private const val CASE = "CASE"
+private const val WHEN = "WHEN"
+private const val THEN = "THEN"
+private const val END = "END"
+private const val ELSE = "ELSE"
+
+sealed interface CaseExpression : Expression
+
+sealed interface UnaliasedCaseExpression : CaseExpression {
+    fun alias(alias: String) = AliasedCaseExpression(alias, this)
+}
+
+class AliasedCaseExpression(
+    private val alias: String,
+    private val unaliasedCaseExpression: UnaliasedCaseExpression,
+) : CaseExpression {
+    override fun toDopeQuery(): DopeQuery {
+        val caseDopeQuery = unaliasedCaseExpression.toDopeQuery()
+
+        val queryString = caseDopeQuery.queryString.trimEnd()
+            .removeSuffix(END)
+            .plus("AS `$alias` $END")
+
+        return DopeQuery(queryString, caseDopeQuery.parameters)
+    }
+}
+
+class CaseClass<T : ValidType>(val case: UnaliasedExpression<T>) : Resolvable {
```

Comment:
Why `CaseClass`? And it's possible to pass just any `UnaliasedExpression`, and thus end up with a potentially incorrect DopeQuery.

Reply:
I named it `CaseClass`, because i couldn't come up with a better name. 
For the Case it is required to be an `UnaliasedExpression` and I can't find an example where it would create invalid syntax. Do you have one?